src/etl/expression_atlas_etl.py

```diff
@@ -0,0 +1,123 @@
+import logging, sys, uuid
+logger = logging.getLogger(__name__)
+
+import multiprocessing
+import xmltodict
+
+from etl import ETL
+from etl.helpers import ETLHelper, Neo4jHelper
+from files import JSONFile
+from transactors import CSVTransactor, Neo4jTransactor
+
+logger = logging.getLogger(__name__)
+
+class ExpressionAtlasETL(ETL):
+    get_all_gene_primary_to_ensmbl_ids_template = """
+        MATCH (g:Gene)-[:CROSS_REFERENCE]-(c:CrossReference)
+        WHERE c.globalCrossRefId STARTS WITH 'ENSEMBL:'
+        RETURN g.primaryKey, c.globalCrossRefId"""
+
+    get_mod_gene_symbol_to_primary_ids_template = """
+        MATCH (g:Gene)
+        WHERE g.dataProvider = {parameter}
+        RETURN g.primaryKey, g.symbol"""
+
+    get_genes_with_expression_atlas_links_template = """
+        MATCH (g:Gene)
+        WHERE LOWER(g.primaryKey) IN {parameter}
+        RETURN g.primaryKey, g.modLocalId"""
+
+    add_expression_atlas_crossreferences_template = """
+        USING PERIODIC COMMIT %s
+        LOAD CSV WITH HEADERS FROM \'file:///%s\' AS row
+
+        MATCH (o:Gene)
+        WHERE o.primaryKey = row.genePrimaryKey """  + ETLHelper.get_cypher_xref_text()
+
+    def __init__(self, config):
+        super().__init__()
+        self.data_type_config = config
+
+    def _load_and_process_data(self):
+        thread_pool = []
+        ensgToGenePrimaryIdMap = self._get_primary_gene_ids_to_ensembl_ids()
+
+        for sub_type in self.data_type_config.get_sub_type_objects():
+            p = multiprocessing.Process(target=self._process_sub_type, args=(sub_type, ensgToGenePrimaryIdMap))
+            p.start()
+            thread_pool.append(p)
+
+        ETL.wait_for_threads(thread_pool)
+
+    def _get_primary_gene_ids_to_ensembl_ids(self):
+        returnSet = Neo4jHelper.run_single_query(ExpressionAtlasETL.get_all_gene_primary_to_ensmbl_ids_template)
+        return {record["c.globalCrossRefId"].split(":")[1].lower(): record["g.primaryKey"] for record in returnSet}
```

Comment:
can you use c.localId instead of splitting the global id again?

Reply:
change made